### DISCUSSION TITLE
Disallowing certain include types

### DISCUSSION THREAD

**Discussion by countless-integers** [spatie/laravel-query-builder#896] [Show and tell]
Right now, adding includes to query params allows for "count" and "exists" variations of them. E.g. 

```php 
$posts = QueryBuilder::for(Post::class)->allowedIncludes(['user']);
```

would accept query params like `include[]=user`, `include[]=userCount` or `include[]=userExists`. This behaviour seems to be a consequence of this bit of code in [`AllowedInclude`](https://github.com/spatie/laravel-query-builder/blob/main/src/AllowedInclude.php#L47). 

What I'd like to achieve is to prevent users from being able to use `userCount` and `userExists` include values (or at the very least, hide those options from the exception message thrown in [`AddsIncludesToQuery`](https://github.com/spatie/laravel-query-builder/blob/main/src/Concerns/AddsIncludesToQuery.php#L81). 

I've somewhat hacked my way to that result by setting suffix values for those parameters to an empty string (e.g. for [`exists_suffix`](https://github.com/spatie/laravel-query-builder/blob/main/config/query-builder.php#L33)), but this feels a bit brittle. 

What would you advise for me to do instead to get that functionality of disallowing those include types?

**Comment by CodeWithDennis** [comment]
**Yes, please!**

I was talking with the owner of [Scramble](https://scramble.dedoc.co/usage/laravel-query-builder) (a Laravel documentation generator) and noticed something strange. When using;

```php
->allowedIncludes(['categories'])`
```

the generated docs also include the `count` and `exists` methods, which didn’t really make sense to me at first. But apparently, that’s what `laravel-query-builder` does.

> Available includes are categories, categoriesCount, categoriesExists. You can include multiple options by separating them with a comma.

It would be nice if there was a way to exclude those two methods.

Thanks to @romalyt for coming up with a quick fix:

```php
collect([new AllowedInclude('categories', new IncludedRelationship())])
```

Still, it would be better if this was handled in the package itself.
